src/peripheral/dwt.rs

```diff
@@ -356,12 +364,26 @@ pub struct ComparatorAddressSettings {
     pub access_type: AccessType,
 }
 
+/// Settings for cycle count matching
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+pub struct CycleCountSettings {
+    /// The function selection used.
+    /// See Table C1-15 for DWT cycle count comparison functions.
+    pub emit: EmitOption,
+    /// The cycle count value to compare against.
+    pub compare: u32,
+    /// The cycle count mask value to use.
+    pub mask: u32,
+}
+
 /// The available functions of a DWT comparator.
 #[derive(Debug, Eq, PartialEq, Copy, Clone)]
 #[non_exhaustive]
 pub enum ComparatorFunction {
     /// Compare accessed memory addresses.
     Address(ComparatorAddressSettings),
+    /// Compare cycle count  & target value.
+    CycleCount(CycleCountSettings),
```

Comment:
```suggestion
    /// Compare cycle count  & target value.
    ///
    /// **NOTE**: only supported by comparator 0. See C1.8.1.
    CycleCount(CycleCountSettings),
```

Reply:
Ideally we should statically check what comparator is being configured, but I don't know whether that is possible.

Reply:
I couldn't think of a way to do so without a more invasive change, like special casing the first two as special comparators. I'm not convinced it'll be really worth it..

Reply:
Added the note